src/basic/time-util.h

```diff
@@ -29,6 +29,11 @@ typedef struct triple_timestamp {
         usec_t boottime;
 } triple_timestamp;
 
+typedef struct optional_usec {
+        usec_t value;
+        bool valid;
+} optional_usec;
```

Comment:
oh, do we really want another structure for this? I mean, this only has a single user. As long as that's the case, let's just keep this in the Service structure...

Reply:
But how should I pass the separate boolean to the parsing function? AFAIK the parser function takes only one pointer to the destination.

BTW, it's not just in the service structure. It's also in `src/core/main.c` as `arg_default_timeout_abort_usec` and the manager structure...

Reply:
Never mind. I just made two functions.